### PR TITLE
Move packaging logic to osquery-packaging

### PR DESCRIPTION
**Pull request by alessandrogario**
This migrates this build-from-source packaging logic into a separate repo.

- The wiki documentation is updated with the new instructions on how to package from source builds.
- The CI is updated to use the new repo for package testing.

### CODE REVIEW

**Comment by theopolis** [comment]
We definitely need to document why this change is happening. And we need to update the docs with instructions on how to build packages.

File: docs/wiki/development/building.md [hunk 1]
```diff
@@ -281,69 +281,123 @@ vagrant up aws-amazon2015.03 --provider=aws
...
-You may override this with the CMake `PACKAGING_SYSTEM` variable as seen in the example below.
+```sh
+cd osquery_source_folder
+git fetch --tags
+
+export OSQUERY_VERSION=$(git describe --tags --always)
+```
+
+**Windows**
+```batch
```
Comment (theopolis) [inline docs/wiki/development/building.md:329]: Double check this file to make sure there are empty newlines between statements and code sections.

File: docs/wiki/development/building.md [hunk 2]
```diff
@@ -281,69 +281,123 @@ vagrant up aws-amazon2015.03 --provider=aws
...
 
 ```sh
-cmake -DOSQUERY_TOOLCHAIN_SYSROOT=/usr/local/osquery-toolchain -DPACKAGING_SYSTEM=RPM ..
-cmake --build . --target package
+git clone https://github.com/osquery/osquery-packaging
+
+mkdir build
+cd build
 ```
+Common input parameters
```
Comment (theopolis) [inline docs/wiki/development/building.md:346]: Empty newline before and after this

File: docs/wiki/development/building.md [hunk 3]
```diff
@@ -281,69 +281,123 @@ vagrant up aws-amazon2015.03 --provider=aws
...
+ - **OSQUERY_VERSION**: can be customized, but we usually use the output of `git describe --always`
+ - **OSQUERY_DATA_PATH**: Where the package data has been installed
 
-### On Windows
+### Creating the Linux packages
 
-On Windows CPack will create an MSI by default. You can toggle this behavior to instead create a
-Chocolatey package by overriding the CMake `PACKAGING_SYSTEM` variable similar to Linux.
+Linux-specific parameters:
```
Comment (theopolis) [inline docs/wiki/development/building.md:352]: Empty newline after this

File: docs/wiki/development/building.md [hunk 4]
```diff
@@ -281,69 +281,123 @@ vagrant up aws-amazon2015.03 --provider=aws
...
+  ../osquery-packaging
+
+cmake --build . --target package
 ```
 
-To instead generate a Chocolatey package, use the following:
+### Creating the Windows packages
+
+Windows-specific parameters:
+ - **CPACK_GENERATOR**: Either `WIX` or `NuGet`
```
Comment (theopolis) [inline docs/wiki/development/building.md:372]: ditto

File: docs/wiki/development/building.md [hunk 5]
```diff
@@ -281,69 +281,123 @@ vagrant up aws-amazon2015.03 --provider=aws
...
+  -DOSQUERY_PACKAGE_VERSION=%OSQUERY_VERSION% ^
+  -DOSQUERY_DATA_PATH=%DESTDIR% ^
+  -DOSQUERY_BITNESS=64 ^
+  ..\osquery-packaging
 
-```sh
-cmake -DPACKAGING_SYSTEM=NuGet -G "Visual Studio 16 2019" -A x64 ..
 cmake --build . --config Release --target package
 ```
+### Creating the macOS packages
```
Comment (theopolis) [inline docs/wiki/development/building.md:389]: ditto before

File: docs/wiki/development/building.md [hunk 6]
```diff
@@ -281,69 +281,123 @@ vagrant up aws-amazon2015.03 --provider=aws
...
 
-### On macOS
-
-On macOS you can choose between a TGZ or a PKG, which is the default.
-You may override this with the CMake `PACKAGING_SYSTEM` variable as seen in the example below.
+macOS-specific parameters:
+ - **CPACK_GENERATOR**: Either `productbuild` (PKG  files) or `TGZ`
```
Comment (theopolis) [inline docs/wiki/development/building.md:392]: ditto

**Review by theopolis — CHANGES_REQUESTED** [review]

**Review by theopolis — DISMISSED** [review]

**Review by directionless — DISMISSED** [review]
At least on macOS, this does produce a `package_data`, which is awesome.

Some nits/suggestions, both of which are not blocking:
1. This build should leave a VERSION file. The packaging consumer should not need the to find the version
2. I want to quibble about the structure in `package_data` -- on macOS, there are 3 directories. `control`, `private`, and `usr`. I'm pretty sure the first is a meta directory, and the latter two are in `/`. I feel like convention would be to create `control` and `root`.

Anyhow, quibbles.

**Review by directionless — APPROVED** [review]

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -359,80 +374,102 @@ jobs:
...
+        done
 
     - name: Locate the packages
       if: matrix.build_type == 'RelWithDebInfo'
-      working-directory: ${{ steps.build_paths.outputs.BINARY }}
-      id: package_names
+      id: packages
       shell: bash
       run: |
-        echo ::set-output name=DEB_PACKAGE_NAME::$(ls *.deb)
```
Comment (theopolis) [inline .github/workflows/build.yml:401]: I think you need to update these variables below (lines 449, etc). I believe this is the cause of the build failures on Linux.

**Review by theopolis — CHANGES_REQUESTED** [review]

**Comment by theopolis** [comment]
This looks good but heads up it has not updated the `build_aarch64` workflows. Those will start to fail after this is merged. Do you want to update them in the same PR or let them break and follow up with another PR.

A good way to test them is to develop in a feature branch on `osquery/osquery`.

**Comment by alessandrogario** [comment]
> This looks good but heads up it has not updated the `build_aarch64` workflows. Those will start to fail after this is merged. Do you want to update them in the same PR or let them break and follow up with another PR.
> 
> A good way to test them is to develop in a feature branch on `osquery/osquery`.

I was wondering how to test this, as I would have preferred to not break the AArch64 workflow; I'll create a new branch and switch my PR to work against it

**Pull request merged**